### ISSUE TITLE
Need a env to configure/disable nat gateway

### ISSUE THREAD

**Issue by rafilkmp3** [cn-terraform/terraform-aws-networking#1]
I would like a way to not create a natgw or create at the max one and route all traffic from private subnet to this single nat_gw, they are expensive

**Comment by jnonino** [comment]
The idea of having one nat gateway per availability zone is for redundancy in case of failure. I'll start working on a solution to match both cases the one is right now and the one you proposed

**Comment by mhmdio** [comment]
Already implemented, PR coming soon.

**Comment by jnonino** [comment]
Released in version [2.0.11](https://registry.terraform.io/modules/cn-terraform/networking/aws/2.0.11)

Thanks to @mhmdio

**Comment by notAmine** [comment]
Could you please provide a simple usage example? 
When I tried this with the following I got an error.
```module "base-network" {
  source     = "cn-terraform/networking/aws"
  cidr_block = "192.168.0.0/16"
  single_nat = true

  vpc_additional_tags = {
    vpc_tag1 = "tag1",
    vpc_tag2 = "tag2",
  }

  public_subnets = {
    first_public_subnet = {
      availability_zone = "eu-west-2a"
      cidr_block        = "192.168.0.0/19"
    }
    second_public_subnet = {
      availability_zone = "eu-west-2b"
      cidr_block        = "192.168.32.0/19"
    }
  }

  public_subnets_additional_tags = {
    public_subnet_tag1 = "tag1",
    public_subnet_tag2 = "tag2",
  }

  private_subnets = {
    first_private_subnet = {
      availability_zone = "eu-west-2a"
      cidr_block        = "192.168.128.0/19"
    }
    second_private_subnet = {
      availability_zone = "eu-west-2b"
      cidr_block        = "192.168.160.0/19"
    }
  }

  private_subnets_additional_tags = {
    private_subnet_tag1 = "tag1",
    private_subnet_tag2 = "tag2",
  }

} 

``` 
> Plan: 19 to add, 0 to change, 0 to destroy.
╷
│ Error: Invalid index
│ 
│   on .terraform/modules/base-network/subnets_private.tf line 49, in resource "aws_route" "private_internet_route":
│   49:   nat_gateway_id = aws_nat_gateway.nat[zipmap(keys(aws_subnet.private), keys(aws_subnet.public))[each.key]].id
│     ├────────────────
│     │ aws_nat_gateway.nat is object with 1 attribute "first_public_subnet"
│     │ aws_subnet.private is object with 2 attributes
│     │ aws_subnet.public is object with 2 attributes
│     │ each.key is "second_private_subnet"
│ 
│ The given key does not identify an element in this collection value.
╵

I am trying to spin up a VPC for the AWS ECS Fargate example here, but I with a single nat gateway for less cost
https://github.com/cn-terraform/terraform-aws-ecs-fargate/blob/main/examples/test/main.tf

Thanks!
